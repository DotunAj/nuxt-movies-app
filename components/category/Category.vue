<template>
  <div class="category">
    <h3>Browse by category</h3>
    <div class="cat-sections">
      <ul>
        <li v-for="genre in genres" :key="genre.id" :class="selected(genre.id)">
          <a @click="changeCategory(genre.id)" v-html="genre.name"></a>
        </li>
      </ul>
    </div>
    <transition name="fade">
      <CategoryItem v-if="!loading" :cat-movies="catMovies" />
    </transition>
    <nuxt-link class="cat__more-link" :to="'/genre/' + genreId"
      >View More</nuxt-link
    >
  </div>
</template>

<script>
import CategoryItem from '~/components/category/CategoryItem'
export default {
  components: {
    CategoryItem
  },
  props: {
    catMovies: {
      type: Array,
      default: () => {}
    },
    changeCategory: {
      type: Function,
      default: () => {}
    },
    loading: Boolean,
    genreId: {
      type: Number,
      default: 28
    }
  },
  data() {
    return {
      genres: [
        { name: '&#128293 Action', id: 28 },
        { name: '&#128081 Adventure', id: 12 },
        { name: '&#127775 Fantasy', id: 14 },
        { name: '&#128150 Romance', id: 10749 }
      ]
    }
  },
  methods: {
    selected(id) {
      let styleClass = ''
      if (this.genreId === id) {
        styleClass = 'selected'
      }
      return styleClass
    }
  }
}
</script>

<style scoped>
.cat-sections a {
  cursor: pointer;
}
</style>
