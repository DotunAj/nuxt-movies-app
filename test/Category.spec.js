import { shallowMount } from '@vue/test-utils'
import Category from '~/components/category/Category'

const catMovies = [
  { id: 1, backdropUrl: 'mybackdrop.com/image1', original_title: 'A new Hope' },
  {
    id: 2,
    backdropUrl: 'mybackdrop.com/image2',
    original_title: 'The rise of the empire'
  },
  {
    id: 3,
    backdropUrl: 'mybackdrop.com/image3',
    original_title: 'I am your father'
  }
]

const genreId = 28

describe('Category', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallowMount(Category, {
      stubs: ['nuxt-link']
    })
    wrapper.setProps({
      catMovies,
      changeCategory: (id) => {
        wrapper.setProps({ genreId: 12 })
      },
      genreId,
      loading: false
    })
  })
  test('it should render the correct content', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('changes style when button is clicked', () => {
    wrapper
      .findAll('a')
      .at(1)
      .trigger('click')
    expect(wrapper.html()).toContain(
      '<li class="selected"><a id="12">ðŸ‘‘ Adventure</a></li>'
    )
    expect(wrapper.html()).not.toContain(
      '<li class="selected"><a id="28">ðŸ”¥ Action</a></li>'
    )
  })
})
