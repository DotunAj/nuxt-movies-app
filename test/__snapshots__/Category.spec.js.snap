// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category it should render the correct content 1`] = `"<div class=\\"category\\"><h3>Browse by category</h3> <div class=\\"cat-sections\\"><ul><li class=\\"selected\\"><a id=\\"28\\">ğŸ”¥ Action</a></li><li class=\\"\\"><a id=\\"12\\">ğŸ‘‘ Adventure</a></li><li class=\\"\\"><a id=\\"14\\">ğŸŒŸ Fantasy</a></li><li class=\\"\\"><a id=\\"10749\\">ğŸ’– Romance</a></li></ul></div> <categoryitem-stub name=\\"fade\\" catmovies=\\"[object Object],[object Object],[object Object]\\"></categoryitem-stub> <nuxt-link-stub to=\\"/genre/28\\" class=\\"cat__more-link\\">View More</nuxt-link-stub></div>"`;
